app/models/course/assessment/question/text_response_comprehension_group.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::TextResponseComprehensionGroup < ApplicationRecord
```

Comment:
Would suggest for Groups, Points and Solution, you write some documentation on how you intend for it to be used, and how the models come together to help solve the issue. This will help other people reading these files to better understand what you are trying to do. 

Suggest you can start with some explanation for the leyman as a first cut.